python/cudf/cudf/io/avro.py

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) 2019, NVIDIA CORPORATION.
+
+import warnings
+
+import pandas as pd
+
+import fastavro as fa
+from cudf.bindings.avro import cpp_read_avro
+from cudf.dataframe.dataframe import DataFrame
+from cudf.utils import ioutils
+
+
+@ioutils.doc_read_avro()
+def read_avro(
+    filepath_or_buffer,
+    engine="cudf",
+    columns=None,
+    skip_rows=None,
+    num_rows=None,
+):
+    """{docstring}"""
+
+    filepath_or_buffer, compression = ioutils.get_filepath_or_buffer(
+        filepath_or_buffer, None
+    )
+    if compression is not None:
+        ValueError("URL content-encoding decompression is not supported")
+
+    if engine == "cudf":
+        df = cpp_read_avro(filepath_or_buffer, columns, skip_rows, num_rows)
+    else:
+        warnings.warn("Using CPU via fastavro to read Avro dataset.")
+        if ioutils.is_file_like(filepath_or_buffer):
+            reader = fa.reader(filepath_or_buffer)
+        else:
+            reader = fa.reader(open(filepath_or_buffer, "rb"))
+
+        pdf = pd.DataFrame.from_records(reader, columns=columns)
```

Comment:
What does avro do with respect to nulls vs NaNs? I imagine this could break things if there's nullable integer columns for example.

Reply:
From what I understand, we only provide this non-cudf path for CI, and test_avro will convert float columns with NaNs to nulls.

Reply:
Actually, I'm thinking that we should probably just not support the non-cudf path here, so as to avoid the fastavro dependency in non-CI path...

Reply:
I agree here.

Reply:
Ok. Replaced the non-cudf engine path by a raise(NotImplementedError)